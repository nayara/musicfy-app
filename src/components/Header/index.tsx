import { FC } from "react";
import * as Style from "./Header.styles";
import Logo from "../Logo";

const Header: FC = () => {
  return (
    <Style.Container>
      <Logo />
    </Style.Container>
  );
};

export default Header;
