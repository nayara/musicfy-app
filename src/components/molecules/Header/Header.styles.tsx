import { styled } from "styled-components";

const Container = styled.header`
  position: sticky;
  top: 0;
  width: 100vw;
  background-color: ${({ theme }) => theme.colors.neutral.black};
  color: ${({ theme }) => theme.colors.neutral.white};
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

export { Container };
