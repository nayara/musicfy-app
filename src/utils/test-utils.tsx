import { RenderOptions, render } from "@testing-library/react";
import { FC, ReactElement, ReactNode } from "react";
import { Providers } from "src/providers";

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <Providers>{children}</Providers>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
