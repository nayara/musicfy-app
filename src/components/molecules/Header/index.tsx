import { FC } from "react";
import * as Style from "./Header.styles";
import Logo from "../Logo";
import LoginButton from "../LoginButton";

const Header: FC = () => {
  return (
    <Style.Container>
      <Logo />
      <LoginButton />
    </Style.Container>
  );
};

export default Header;
