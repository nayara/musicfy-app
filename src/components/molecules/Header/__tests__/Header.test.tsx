import { render } from "../../../../utils/test-utils";
import Header from "..";

jest.mock("react-router-dom");

describe("<Header />", () => {
  it("should render header correcly", () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
