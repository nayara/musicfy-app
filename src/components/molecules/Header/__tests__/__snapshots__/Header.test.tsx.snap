// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render header correcly 1`] = `
.c0 {
  position: sticky;
  top: 0;
  width: 100vw;
  background-color: #090B0B;
  color: #EEF1EF;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.c2 {
  font-family: "Genty";
  font-size: 3vh;
  padding: 1.5vh;
  margin: 0;
}

.c1 {
  background: none;
  color: inherit;
  border: none;
  padding: 0.5rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
}

.c3 {
  font-size: 2vh;
  color: #EEF1EF;
  padding: 0.5rem 1rem;
  background-color: #c1292e;
  border-radius: 0.5rem;
  margin: 20px;
  font-weight: bold;
  letter-spacing: 0.01rem;
}

<div>
  <header
    class="c0"
  >
    <button
      class="c1"
    >
      <h1
        class="c2"
      >
        MusicFy
      </h1>
    </button>
    <button
      class="c1"
    >
      <span
        class="c3"
      >
        Login
      </span>
    </button>
  </header>
</div>
`;
